### ISSUE TITLE
Tree is not displayed on macOS 10.13.6

### ISSUE THREAD

**Issue by TempSpas** [piroor/treestyletab#1979]
## Short description

I updated my computer from 10.13.5 to 10.13.6 today and the extension no longer displays tabs at all. Every window I open shows the loading icon on the sidebar forever. It also triggers the computer's fan to go off. I have tried restarting Firefox, toggling the sidebar, and disabling/re-enabling the extension. None of these steps solved the problem.

## Steps to reproduce

 1. Open Firefox on macOS 10.13.6 and restore previous session.
 2. Observe tabs are not displayed and sidebar loading icon appears endless.
 3. Open a new window.
 4. Observe the same issue in the new window.

## Expected result

Tabs should load and display normally.

## Actual result

Loading icon displays forever and tabs are never displayed. Fan is triggered and computer heats up.

<img width="166" alt="screen shot 2018-07-11 at 11 48 51 am" src="https://user-images.githubusercontent.com/3801275/42584296-df9e89ba-8500-11e8-9ad0-85323aaf28d9.png">

## Environment

 * Platform (OS): macOS 10.13.6
 * Version of Firefox: 61.0.1
 * Version (or revision) of Tree Style Tab: 2.4.24

**Comment by TempSpas** [comment]
Completely uninstalling the addon and reinstalling it fixed the loading issue.

**Comment by piroor** [comment]
OK, I close this for now. If you see the problem again with [the latest development build of TST](https://github.com/piroor/treestyletab/blob/master/README.md#development-builds), please post logs exported via the remote debugger console](https://github.com/piroor/treestyletab/issues/1735#issuecomment-392656263).

**Comment by TempSpas** [comment]
I experienced a browser crash when attempting to load CSPAN, and now the extension seems completely broken. I have tried uninstalling and reinstalling it, I have tried uninstalling it, restarting my computer, and reinstalling it, and the issue persists. Sidebar is stuck infinitely loading, and nothing I do can fix it.

Won't be able to test the latest development build for at least a month. In the meantime, this is the result of running `gMetricsData.toString()`:

<img width="311" alt="screen shot 2018-07-12 at 5 17 18 pm" src="https://user-images.githubusercontent.com/3801275/42659907-acd39406-85f7-11e8-9405-3a9cfd75feb4.png">

**Comment by piroor** [comment]
The log says that the sidebar is waiting for the completion of the background page's initialization. Could you try `gMetricsData.toString()` for the `background/background.html`?

**Comment by TempSpas** [comment]
<img width="1012" alt="screen shot 2018-07-14 at 3 10 13 pm" src="https://user-images.githubusercontent.com/3801275/42727603-1925e592-8778-11e8-94aa-3f15a9cc04ee.png">

**Comment by piroor** [comment]
Hmm, then, the problem is caused by a broken cache. Disabling cache mechanism and re-enabling will clear such a broken cache, so it should solve it. (It is just a workaround, and more error handling for broken cache must be added in future versions.)

**Comment by TempSpas** [comment]
I unchecked and checked the box, but it didn't fix the issue. The sidebar showed no change. Interestingly, after checking and unchecking that box, I can no longer click the Tree Style Tab toolbar icon to toggle the sidebar.

**Comment by piroor** [comment]
Sorry I misread the log. It said that there are 7 windows but failed to initialize for one of them. However, it is still unclear that why it failed. Could you collect more logs? To activate logging, you'll need to activate "Debug mode", check "background/background.js" and "background/background-cache.js", and uncheck other log modules.

**Comment by TempSpas** [comment]
I don't see either of those options available to check off under Debug mode, so you're going to have to tell me exactly where those are. This is all I see: 
<img width="694" alt="screen shot 2018-07-23 at 9 56 30 am" src="https://user-images.githubusercontent.com/3801275/43080931-d91b8a2c-8e5e-11e8-8876-23030b5e4b0d.png">

As you can see, I checked off the "cached tree" box just because it was the only option that shared a word with the ones you suggested. I've attached the log file that outputted.

[tst_log.txt](https://github.com/piroor/treestyletab/files/2219567/tst_log.txt)

**Comment by piroor** [comment]
Thank you for collecting logs, and I've realized that there is a big confusion. Currently a large reconstruction is in-progress on the development version of TST (it will be released as 2.5), thus problems reported for 2.4.24 need to be re-examined. Very important information is:

 1. Does the problem still happen on the latest development build?
 2. If it still happens, what logs appears on the latest development build?

There is [instructions to try the latest development build](https://github.com/piroor/treestyletab/blob/master/README.md#development-builds). Safe steps to re-examine the problem you saw with the development build are:

 1. Open the active profile folder from "about:support".
 2. Clone the folder (for re-examining).
 3. Download Nightly and install it.
 4. Start Nightly with options: `-profile /path/to/cloned/profile -no-remote`.
    See also: [how to start Firefox with command line options on macOS](https://developer.mozilla.org/en-US/docs/Mozilla/Command_Line_Options#Mac_OS_X).

**Comment by xarx00** [comment]
I have the same problem, though I am on Windows, not on MAC. Sometimes - it happened twice to me in the last couple of days - the Tree Style Tab sidebar did not initialize. It ended in the same way as in the original report https://github.com/piroor/treestyletab/issues/1979#issue-340303638.

I have tried to find out what other addon - if any - causes this problem, unfortunately I'm not able to simulate the problem by request. While I was experimenting with various settings, the problem did exhibit not even once.

If this problem is caused not by TST itself ,but instead by a feature in FF or by another addon, my guesses are:
* I am using Auto Tab Discard to hibernate unused tabs; hence the problem could be related to #1984. Nevertheless, even when I discarded all tabs, TFT always initialized properly.
* I am using containers, in particular Firefox Multi-Account Containers and Temporary Containers addons; hence the problem may be caused by an attempt of one of the addons to move a tab from one container to a container specified by rules. Containers are a FF feature, the addons only make a UI over this functionality, so the tabs are being restored to containers even when the addons are disabled.
* I'm using the Tab Session Manager addon; however, I don't think it is involved while FF is started in a normal way. (In TSM addon settings I have "Restore previous session on startup" turned off.)

Windows 7 Pro x64
FF 60.1.0 ESR x64
TST 2.4.24
Auto Tab Discard 0.2.8
Firefox Multi-Account Containers 6.0.0
Temporary Containers 0.90
Tab Session Manager 3.2.0

**Comment by piroor** [comment]
Anyway I need very narrowed and certain conditions to reproduce this problem on my environment for further research. I cannot take much time to digging trigger of such difficult problems for now, so pull requests fixsing them are really needed...

**Comment by piroor** [comment]
You need to try to run on remote addon debugger started via about:debugging, instead of regular debugger. And you may need to choose "background/background.html" from the list of inspected targets. Sometimes "options/options.html" and other pages are selected by default.

**Comment by piroor** [comment]
TST's logs around startup process are stored if you activate TST's "Debug mode":
![image](https://user-images.githubusercontent.com/70062/43442649-8ca422fe-94d9-11e8-900e-9f98a112b555.png)

And the list of target windows is available here:
![image](https://user-images.githubusercontent.com/70062/43442717-bbeeb2b8-94d9-11e8-96bd-17d001c8387c.png)

**Comment by Andrade** [comment]
I had this issue but I'm still on macOS 10.13.5. Disabling TST cache and restarting browser solved it.

_Avanced_ → _Optimize tree restoration with cache_

**Comment by xarx00** [comment]
The problem hasn't occurred in its full severity since my report. However, I've noticed a strange behaviour of TST, which ultimately may have caused the described problem.

When I right-click an item in the tree and choose DuplicateTab, TST shows the "running circle". Until the "running circle" disappears, I cannot click anywhere into the TST tree. This state usually lasts for a couple of seconds (approx 3s). I have no idea why TST freezes for so long, perhaps parts of the page are unreachable? But this happens on any page I have tried, even when I turn off my ad-blocker.

Maybe the same behaviour could have cased the initial problem. If, say, I had several unreachable pages in my tree upon  FF restart, TST could have shown the "running circle" for a pretty long time to look as frozen.
